### ISSUE TITLE
tenant doing incorrent number of retries during the verification 

### ISSUE THREAD

**Issue by kkaarreell** [keylime/keylime#1054]
Is your an issue a feature request? If so, please raise it as an [enhancement](https://github.com/keylime/enhancements)

# Environment
 * OS / version: CentOS Stream 9
 * Processor architecture: s390x (but IMHO should be arch independent)
 * TPM Manufacturer: swtpm
 * Keylime version: current upstream 8b8d0c60da066bd55fe05e3399d2ea63d3d00652

# Description 

```
# keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u d432fbb3-d2f1-4a97-9ef7-75bd81c00000 --verify --allowlist allowlist.txt --exclude excludelist.txt --include payload-module --cert default -c add
Reading configuration from ['/etc/keylime.conf']
2022-07-21 09:40:37.910 - keylime.tpm - INFO - TPM2-TOOLS Version: 5.2
2022-07-21 09:40:37.912 - keylime.tenant - INFO - Setting up client TLS in /var/lib/keylime/cv_ca
2022-07-21 09:40:37.912 - keylime.registrar_client - WARNING - TLS is enabled.
2022-07-21 09:40:37.912 - keylime.registrar_client - INFO - Setting up client TLS...
2022-07-21 09:40:37.950 - keylime.tenant - INFO - TPM PCR Mask from policy is 0x408000
Please enter the password to decrypt your keystore: 
2022-07-21 09:40:38.016 - keylime.ca-util - INFO - Creating cert package for d432fbb3-d2f1-4a97-9ef7-75bd81c00000 in d432fbb3-d2f1-4a97-9ef7-75bd81c00000-pkg.zip
2022-07-21 09:40:38.071 - keylime.ca-util - INFO - Creating cert package for RevocationNotifier in RevocationNotifier-pkg.zip
<Response [200]>
2022-07-21 09:40:38.180 - keylime.tpm - WARNING - PCR #0 in quote not found in tpm_policy, skipping.
2022-07-21 09:40:38.181 - keylime.tenant - WARNING - DANGER: EK cert checking is disabled and no additional checks on EKs have been specified with ek_check_script option. Keylime is not secure!!
2022-07-21 09:40:38.181 - keylime.tenant - INFO - Quote from 127.0.0.1 validated
2022-07-21 09:40:38.324 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:38.324 - keylime.tenant - INFO - Key derivation not yet complete (retry 1/5), trying again in 2.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:40.346 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:40.346 - keylime.tenant - INFO - Key derivation not yet complete (retry 2/5), trying again in 4.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:44.373 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:44.373 - keylime.tenant - INFO - Key derivation not yet complete (retry 3/5), trying again in 8.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:52.402 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:52.402 - keylime.tenant - INFO - Key derivation not yet complete (retry 4/5), trying again in 16.0 seconds... (Ctrl-C to stop)
:: [ 09:40:57 ] :: [   PASS   ] :: Command 'expect script.expect' (Expected 0, got 0)
```

In /etc/keylime.conf in [tenant] section I have
```
# Number of retries to connect to a agent before giving up. Must be an integer.
max_retries = 5
```

## Expected behavior vs. actual behavior 
 There should be one more iteration, i.e. 6 in total - 1st attempt + 5 retries
followed by the error message.

## Steps to reproduce problem
Make the tenant verification fail somehow.

# Relevant logs

Attach any relevant log files that can help to debug your issue. 

```
2022-07-21 09:40:38.181 - keylime.tenant - WARNING - DANGER: EK cert checking is disabled and no additional checks on EKs have been specified with ek_check_script option. Keylime is not secure!!
2022-07-21 09:40:38.181 - keylime.tenant - INFO - Quote from 127.0.0.1 validated
2022-07-21 09:40:38.324 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:38.324 - keylime.tenant - INFO - Key derivation not yet complete (retry 1/5), trying again in 2.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:40.346 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:40.346 - keylime.tenant - INFO - Key derivation not yet complete (retry 2/5), trying again in 4.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:44.373 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:44.373 - keylime.tenant - INFO - Key derivation not yet complete (retry 3/5), trying again in 8.0 seconds... (Ctrl-C to stop)
2022-07-21 09:40:52.402 - keylime.tenant - ERROR - Response code 400: Bootstrap key not yet available.
2022-07-21 09:40:52.402 - keylime.tenant - INFO - Key derivation not yet complete (retry 4/5), trying again in 16.0 seconds... (Ctrl-C to stop)


keylime_verifier[13545]: 2022-07-21 09:40:38.484 - keylime.ima - ERROR - IMA ERRORS: Some entries couldn't be validated. Number of failures in modes: ImaNg 633.
```

**Comment by kkaarreell** [comment]
It may be in fact a bit more difficult. Since I would at least expect some ERROR to be printed after the last attempt but what happens here is that keylime_tenant ends ~5 seconds after its intent to way for another 16 seconds is printed.

**Comment by ansasaki** [comment]
I believe this was fixed by #1271
